src/main/java/net/sky/chess/Board.java

```diff
@@ -3,107 +3,159 @@
     }
 
-    public void addBlackPawn(Piece piece) {
-        blackPawns.add(piece);
+    public void initializeEmpty() {
+        for (int rank = START_RANK; rank >= END_RANK; rank--) {
+            initializeBlanks(rank);
+        }
+
     }
 
-    public void addWhitePiece(Piece piece) {
-        whitePieces.add(piece);
+    private void initializePieces(int rank, PieceMaker pieceMaker) {
+        char file = START_FILE;
+
+        piecePositions.put(new Position(rank, file++), pieceMaker.createRook());
+        piecePositions.put(new Position(rank, file++), pieceMaker.createKnight());
+        piecePositions.put(new Position(rank, file++), pieceMaker.createBishop());
+        piecePositions.put(new Position(rank, file++), pieceMaker.createQueen());
+        piecePositions.put(new Position(rank, file++), pieceMaker.createKing());
+        piecePositions.put(new Position(rank, file++), pieceMaker.createBishop());
+        piecePositions.put(new Position(rank, file++), pieceMaker.createKnight());
+        piecePositions.put(new Position(rank, file), pieceMaker.createRook());
     }
 
-    public void addBlackPiece(Piece piece) {
-        blackPieces.add(piece);
+    private void initializePawns(int rank, PieceMaker pieceMaker) {
+        for (char file = START_FILE; file <= END_FILE; file++) {
+            piecePositions.put(new Position(rank, file), pieceMaker.createPawn());
+        }
     }
 
-    public void initialize() {
-        initializeWhitePawns();
-        initializeBlackPawns();
-        initializeWhitePieces();
-        initializeBlackPieces();
+    private void initializeBlanks(int rank) {
+        for (char file = START_FILE; file <= END_FILE; file++) {
+            piecePositions.put(new Position(rank, file), PieceMaker.createBlank());
+        }
+    }
+
+    public void print() {
+        System.out.println(showBoard());
     }
 
-    private void initializeWhitePawns() {
-        for (int i = 0; i < 8; i++) {
-            addWhitePawn(whitePieceMaker.createPawn());
+    public String showBoard() {
+        StringBuilder result = new StringBuilder();
+        final int FILE_SIZE = 8;
+        int initRank = START_RANK;
+        int resultSize = 0;
+
+        for (Piece piece : piecePositions.values()) {
+            resultSize++;
+            result.append(piece.getRepresentation());
+            if (resultSize % FILE_SIZE == 0) {
+                result.append(appendNewLine(" " + initRank--));
+            }
         }
+        result.append(appendNewLine(""));
+        result.append(appendNewLine("abcdefgh"));
+
+        return result.toString();
     }
 
-    private void initializeBlackPawns() {
-        for (int i = 0; i < 8; i++) {
-            addBlackPawn(blackPieceMaker.createPawn());
+    public int pieceCount(Piece piece) {
+        int pieceCount = 0;
+
+        for (Piece p : piecePositions.values()) {
+            if (p.equals(piece)) {
+                pieceCount += 1;
+            }
```

Comment:
`stream` 으로 개선할 수 있을 것 같은데 어떨까요? 

```suggestion
        placePositions.values().stream().filter(p -> p.equals(piece)).count()

Reply:
훨씬 직관적인 것 같아서 좋은 것 같아요. 변경 해볼게요!